IM_calculation/IM/intensity_measures.py

```diff
@@ -31,15 +31,46 @@ def get_spectral_acceleration(acceleration, period, NT, DT):
 def get_spectral_acceleration_nd(acceleration, period, NT, DT):
     # pSA
     if acceleration.ndim != 1:
-        dims = acceleration.shape[1]
+        ts, dims = acceleration.shape
         values = np.zeros((period.size, dims))
+        displacements = np.zeros((period.size, ts * int(round(DT / 0.005)), dims))
         for i in range(dims):
-            values[:, i] = get_spectral_acceleration(acceleration[:, i], period, NT, DT)
-        return values
+            psa, u = get_spectral_acceleration(acceleration[:, i], period, NT, DT)
```

Comment:
I thought this was vectorised. Apparently not. Ohhhh It's vectorised across periods though.